Software/vortex_ws/src/sensors/src/dvl_serial.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright 2021 VorteX-co
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <string>
+#include <vector>
+#include "../include/dvl_companion/linux_serial.hpp"
+#include "../include/dvl_companion/parser.hpp"
+#include "custom_ros_interfaces/msg/dvl.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+using std::placeholders::_1;
+
+Serial serial;  // Serial port instance
+
+class Publisher : public rclcpp::Node
+{
+public:
+  inline Publisher();
+  inline void publish(std::string packet);
+
+private:
+  rclcpp::Publisher<custom_ros_interfaces::msg::DVL>::SharedPtr publisher_;
+};
+Publisher::Publisher()
+: Node("a50_serial_pub")
+{
+  this->declare_parameter<std::string>("Port", "/dev/ttyUSB1");
+  // declares a rclcpp param called Port with a default value
+  this->declare_parameter<std::int64_t>("Baud_Rate", 115200);
+  std::string topic = "dvl_report";
```

Comment:
cpplint imposes that: the line's characters must be <= 80 , so i declared a redundant variable called topic